src/interfaces/highs_csharp_api.cs

```diff
@@ -130,173 +131,198 @@ public HighsBasis(HighsBasisStatus[] colbasisstatus, HighsBasisStatus[] rowbasis
-   private static extern void Highs_destroy(void* highs);
+   private static extern void Highs_destroy(IntPtr highs);
 
    [DllImport(highslibname)]
-   private static extern int Highs_run(void* highs);
+   private static extern int Highs_run(IntPtr highs);
 
    [DllImport(highslibname)]
-   private static extern int Highs_readModel(void* highs, string filename);
+   private static extern int Highs_readModel(IntPtr highs, string filename);
 
    [DllImport(highslibname)]
-   private static extern int Highs_writeModel(void* highs, string filename);
+   private static extern int Highs_writeModel(IntPtr highs, string filename);
 
    [DllImport(highslibname)]
-   private static extern int Highs_passLp(void* highs, int numcol, int numrow, int numnz, int aformat, int sense, double offset, double[] colcost,
+   private static extern int Highs_passLp(IntPtr highs, int numcol, int numrow, int numnz, int aformat, int sense, double offset, double[] colcost,
    double[] collower, double[] colupper, double[] rowlower, double[] rowupper, int[] astart, int[] aindex, double[] avalue);
 
    [DllImport(highslibname)]
-   private static extern int Highs_setOptionValue(void* highs, string option, string value);
+   private static extern int Highs_setOptionValue(IntPtr highs, string option, string value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getSolution(void* highs, double[] colvalue, double[] coldual, double[] rowvalue, double[] rowdual);
+   private static extern int Highs_setBoolOptionValue(IntPtr highs, string option, int value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getNumCol(void* highs);
+   private static extern int Highs_setIntOptionValue(IntPtr highs, string option, int value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getNumRow(void* highs);
+   private static extern int Highs_setDoubleOptionValue(IntPtr highs, string option, double value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getNumNz(void* highs);
+   private static extern int Highs_setStringOptionValue(IntPtr highs, string option, string value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getBasis(void* highs, int[] colstatus, int[] rowstatus);
+   private static extern int Highs_getBoolOptionValue(IntPtr highs, string option, out int value);
 
    [DllImport(highslibname)]
-   private static extern double Highs_getObjectiveValue(void* highs);
+   private static extern int Highs_getIntOptionValue(IntPtr highs, string option, out int value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getIterationCount(void* highs);
+   private static extern int Highs_getDoubleOptionValue(IntPtr highs, string option, out double value);
 
    [DllImport(highslibname)]
-   private static extern int Highs_getModelStatus(void* highs);
+   private static extern int Highs_getStringOptionValue(IntPtr highs, string option, [Out] StringBuilder value);
```

Comment:
can't use string here because HiGHS is writing to the parameter.

Reply:
Sorry, but I understand little of this. I see that we've no getters and setters for the options, and maybe you indicate that we should have `Highs_getStringOptionValue`. But why delete `Highs_getModelStatus`? Also, are you saying that `string` can't be used in the parameter list of `Highs_getStringOptionValue` - that you created?

Unfortunately, no-one in the HiGHS team is a C# user, and we have little Windows expertise.

Reply:
Highs_getModelStatus is still there. Only "void*" has been replaced with IntPtr causing git to show the line as deleted and newly added (scroll down...).

The problem is that the internal representation of string in C# is incompatible with C-style 0-terminated strings. The .NET Runtime therefore has to 'marshall' string parameters. If a string is passed into a function this works by allocating a temporary char* buffer of suitable size, copying the content into that and then pass the buffer to C. All this happens automatically.

Furthermore strings in C# are immutable. Therefore it is impossible to let a C function manipulate the string without risking memory corruption. 
The StringBuilder is mutable and therefore usable if the C function is manipulating the string - the .NET Runtime can simply copy the contents of the char* buffer back into the StringBuilder after the function has returned.
It is not visible to the user anyway.

Reply:
> Highs_getModelStatus is still there. Only "void*" has been replaced with IntPtr causing git to show the line as deleted and newly added (scroll down...).

OK, that's clear enough. I now see [478d3c6](https://github.com/ERGO-Code/HiGHS/pull/764/commits/478d3c66de93155c864d405b7b268ad793c9d40d) and realise that you've made all the necessary changes. 

I'm happy to approve the changes to the C# API

As for the discussion on libraries, again it means nothing to me as I'm a very limited Windows user. However, since you're only changing the C#API I guess that these issues are still to be addressed.